### PR TITLE
Lean tweaks

### PR DESCRIPTION
**Pull request by namin**

### CODE REVIEW

**Comment by namin** [comment]
@johnjyang -- can you review?

File: prompts.py [hunk 1]
```diff
@@ -1,18 +1,19 @@
...
+check_proof = None
 if lang == 'Dafny':
     proof_marker = 'ensures'
 elif lang == 'Coq':
     proof_marker = 'Qed'
 elif lang == 'Lean4':
-    proof_marker = None
-else:
-    proof_marker = None
+    check_proof = lambda v: v.count('theorem') > 2 # because of example
```
Comment (johnjyang) [inline prompts.py:10]: I copied the example from https://github.com/yangky11/lean4-example/blob/main/Lean4Example.lean, it contains two proofs for the sake of it, but most proofs should only have 1 "theorem." I think a check for "theorem" in "v" should work (provided we limit the problems to proofs)!

Comment (namin) [reply prompts.py:10]: Indeed, so > 2 means 1 new theorem!

File: prompts.py [hunk 2]
```diff
@@ -100,5 +95,22 @@
 """, 1000, None, 5, check_proof, ['Dafny'])
 
 # Set the right-hand side to the selected problem.
-(prompt, max_new_tokens, expansion_count, min_lines, check_fun, supported_langs) = problem_intersperse_dafny
+(prompt, max_new_tokens, expansion_count, min_lines, check_fun, supported_langs) = problem_fact
 assert lang in supported_langs
+if lang == 'Lean4':
+    prompt += '''
```
Comment (johnjyang) [inline prompts.py:101]: This is adding the string to the end of the prompt: we can do `prompt = prompt.replace(f"### {lang}:", custom + f"\n### {lang}:")`? Or keeping it where it was.

File: prompts.py (hunk 1 above)
Comment (johnjyang) [reply prompts.py:10]: Oh great! This should work then.

File: prompts.py (hunk 2 above)
Comment (namin) [reply prompts.py:101]: Or just `prompt = custom + "\n" + prompt?`

Comment (johnjyang) [reply prompts.py:101]: Ok!

Thank you!

**Pull request merged**